File: java-to-go-projects/14-http-proxy/Logger.java
package proxy;

import java.time.Instant;
import java.time.Duration;

/**
 * Logs incoming requests and outgoing responses with timing.
 * Formats output for console display.
 */
public class Logger {

    private boolean enabled;

    public Logger(boolean enabled) {
        this.enabled = enabled;
    }

    public void logRequest(String method, String path, String clientIp) {
        if (!enabled) return;
        System.out.printf("[%s] → %s %s (from %s)%n",
                Instant.now(), method, path, clientIp);
    }

    public void logResponse(String method, String path, int statusCode, long durationMs) {
        if (!enabled) return;
        System.out.printf("[%s] ← %s %s → %d (%dms)%n",
                Instant.now(), method, path, statusCode, durationMs);
    }

    public void logError(String method, String path, String error) {
        System.err.printf("[%s] ✗ %s %s → ERROR: %s%n",
                Instant.now(), method, path, error);
    }

    public void setEnabled(boolean enabled) { this.enabled = enabled; }
}

File: java-to-go-projects/14-http-proxy/rust/src/logger.rs
use std::time::Instant;

/// Log an incoming request (method, path, client IP).
pub fn log_request(method: &str, path: &str, client_ip: &str) {
    todo!()
}

/// Log a proxied response (status, duration).
pub fn log_response(status: u16, duration: &Instant) {
    todo!()
}

/// Log a proxy error.
pub fn log_error(path: &str, error: &str) {
    todo!()
}
